src/Microsoft.DotNet.Build.CloudTestTasks/SendJsonToDocumentDb.cs

```diff
@@ -0,0 +1,167 @@
+            using (JsonTextReader jsonReader = new JsonTextReader(streamReader))
+            {
+                var jsonSerializer = new JsonSerializer();
+                json = (JObject)jsonSerializer.Deserialize(jsonReader);
+
+                // add the document ID to the JSON blob
+                json.Add("id", DocumentId);
+            }
+
+            using (var con = CreateDocumentDbConnection())
+            {
+                // for specified document ID check if there is a document for it, if there
+                // isn't then create a new one with the JSON blob that was loaded earlier.
+
+                var result = con.Client.CreateDocumentQuery(string.Format("dbs/{0}/colls/{1}", con.Database.Id, con.Collection.Id))
+                    .Where(doc => doc.Id == DocumentId)
+                    .AsEnumerable()
+                    .FirstOrDefault();
+
+                if (result == null)
+                {
+                    Log.LogMessage("Creating new document with ID '{0}'", DocumentId);
+                    using (var task = con.Client.CreateDocumentAsync(con.Collection.SelfLink, json))
+                        task.Wait();
+                }
+                else
+                {
+                    Log.LogWarning("A document with ID '{0}' already exists, no content was uploaded.", DocumentId);
+                }
+            }
+
+            return true;
+        }
+
+        private class DocumentConnection : IDisposable
+        {
+            private Database m_db;
+            private DocumentClient m_client;
+            private DocumentCollection m_collection;
+           
+            public DocumentConnection(Database db, DocumentClient client, DocumentCollection collection)
+            {
+                m_db = db;
+                m_client = client;
+                m_collection = collection;
+            }
+
+            public Database Database { get { return m_db; } }
+
+            public DocumentClient Client { get { return m_client; } }
+
+            public DocumentCollection Collection { get { return m_collection; } }
+
+            public void Dispose()
+            {
+                m_client.Dispose();
+            }
+        }
+
+        private DocumentConnection CreateDocumentDbConnection()
+        {
+            var client = new DocumentClient(new Uri(EndpointUri), AccountKey);
+
+            // get the database and if it doesn't exist create it
+
+            Database database = client.CreateDatabaseQuery()
+                .Where(db => db.Id == Database)
+                .AsEnumerable()
+                .FirstOrDefault();
+
+            if (database == null)
+            {
+                Log.LogMessage(MessageImportance.Low, "The database {0} does not exist, will create it.", Database);
+                using (var task = client.CreateDatabaseAsync(new Database { Id = Database }))
+                {
+                    task.Wait();
+                    database = task.Result;
```

Comment:
Result is already blocking, so calling Wait() before it doesn't actually do anything, to my knowledge.